mrmustard/lab_dev/circuit_components.py

```diff
@@ -602,9 +609,39 @@ def to_bargmann(self) -> CircuitComponent:
             >>> assert d_bargmann.wires == d.wires
             >>> assert isinstance(d_bargmann.representation, Bargmann)
         """
-        if isinstance(self.representation, Bargmann):
+        from .circuit_components_utils import BtoQ, BtoPS
+        if isinstance(self.representation, Bargmann):# TODO: better name for Bargmann class
+            # check cc rep
+            if not indices:
+                indices = self.wires.indices
+            
+            for i in indices:
+                name, arg =  self._index_representation[i]
+                if name == 'Q':
+                    self._index_representation[i] = ('B', None)
+                    if i in self.wires.output.bra.indices:
+                        self = self @ BtoQ([self.wires.index_to_mode_dict[i]], phi = arg).adjoint.inverse()
+                    if i in self.wires.output.ket.indices:
+                        self = self @ BtoQ([self.wires.index_to_mode_dict[i]], phi = arg).inverse()
+                    if i in self.wires.input.bra.indices:
+                        self = BtoQ([self.wires.index_to_mode_dict[i]], phi = arg).dual.adjoint.inverse() @ self
+                    if i in self.wires.input.bra.indices:
+                        self = BtoQ([self.wires.index_to_mode_dict[i]], phi = arg).dual.inverse() @ self 
+                    
+
+                if name == 'PS':
```

Comment:
You can almost certainly reduce these to a single block e.g.
```suggestion
               bargmann_conversion = BtoQ(...) if name=='Q' else BtoPS() 
	       if i in self.wires.output.bra.indices:
               		ret = self @ bargmann_conversion.adjoint.inverse()
```
Also you shouldn't be overriding self but instead assigning to a return variable